# Oviz-Bio-demo

Documents of [Oviz-Bio](https://bio.oviz.org/), a Platform for Bioinformatics Visualization

# Publication

Jia, Wenlong, Hechen Li, Shiying Li, Lingxi Chen, and Shuai Cheng Li. "[Oviz-Bio: a web-based platform for interactive cancer genomics data visualization.](https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkaa371)" Nucleic Acids Research (2020).

# Intergrated

## LandScape
The `LandScape` visualization is frequently utilized to provide a systematic illustration of integrative data from multiple layers of batch samples, which are always compared to each other on certain attributes, such as genes and biological pathways mutated in cancers. To visualize data, a CSV file in the required format is needed.
### demo_data
- Check the official demo input [here](https://github.com/Nobel-Justin/Oviz-Bio-demo/blob/master/LandScape/demo_data/).
- **PMID** prefixed CSV files to present LandScape instances of published papers (see [Reference](https://github.com/Nobel-Justin/Oviz-Bio-demo/blob/master/LandScape/markdown/LandScape-References.markdown)).

# Small Mutation

## Mut: On Genes
The `Mutations on Genes` displays SNVs and InDels with their coordinates and function annotations along the selected transcript of a given gene in three levels (genome, cDNA, and peptide).
### demo_data
- Check the official demo input [here](https://github.com/Nobel-Justin/Oviz-Bio-demo/blob/master/Mut_OnGenes/demo_data/).

## SNV: Context
The `SNV: Context` visualization, otherwise known as the **Lego plot** of mutational frequencies, describes the distribution of mutations across batch samples on a given region.
### demo_data
- Check the official demo input [here](https://github.com/Nobel-Justin/Oviz-Bio-demo/blob/master/SNV_Context/demo_data/).

## SNV: Signature
The `SNV: Signature` visualization is wildly used in mutational signature analysis. It shows the signature profile with the conventional 96 mutation type classification. We also compare each signature with all recorded signature references in the [COSMIC database](https://cancer.sanger.ac.uk/cosmic/signatures/SBS/) by the cosine similarity test. The top three similar signature references are listed with their percentage of similarity.
### demo_data
- Check the official demo input [here](https://github.com/Nobel-Justin/Oviz-Bio-demo/blob/master/SNV_Signature/demo_data/).

## Mut: Signature Dist
The `Mut: Signature Dist` visualization shows the fraction of signatures within individual samples. It can also be used to show the fraction of signatures within several cancer types.
### demo_data
- Check the official demo input [here](https://github.com/Nobel-Justin/Oviz-Bio-demo/blob/master/Mut_Signature_dist/demo_data/).

# Copy Number Variation

## CNV: Haplotype View
The `CNV: Haplotype View` visualization is an interactive version of the output plot from `Patchwork`, a bioinformatics tool for analyzing allele-specific copy numbers and loss-of-heterozygosity in cancer genomes. This visualization summarizes all 24 chromosomal plots generated by `Patchwork` and provides user with the choice to highlight information of one particular chromosome.
### demo_data
- Check the official demo input [here](https://github.com/Nobel-Justin/Oviz-Bio-demo/blob/master/CNV_Haplotype_View/demo_data/).
- User can directly use the output from the `Patchwork` tool as our input file.

## CNV: Focal Cluster
The `CNV: Focal Cluster` visualization is an interactive and extended version of the output plot from `GISTIC`, a bioinformatics tool for identifying regions of the genome that are significantly amplified or deleted across a set of samples.
### demo_data
- Check the official demo input [here](https://github.com/Nobel-Justin/Oviz-Bio-demo/blob/master/CNV_Focal_Cluster/demo_data/).
- User can directly use the output from the `GISTIC` tool as our input file.

# Structure Variation

## SV: Reads Support
The `SV: Reads Support` visualization is frequently utilized to provide the detail illustration of supporting split reads for given SV events. To visualize data, a **TXT** file in the required format is needed.
### demo_data
- Check the official demo input [here](https://github.com/Nobel-Justin/Oviz-Bio-demo/blob/master/SV_Read_Support/demo_data/).

## SV: Heatmap
Linkage heatmap is an useful method to visualize read linkage patterns on SV event, especially in studies with long-range sequencing data, such as 10x linked-reads. We apply the `SV: Heatmap` visualization to display the heatmap matrix based on sequencing reads linkage between two local windowlized regions of SV case. The color depth of each cross-linked window pair is proportional to the number of linkages. Along the chromosome coordinate axis of the heatmap matrix, annotation information is added, such as Ensembl genes. To visualize data, upload a **TXT** file in the *required* format, and then use sidebar options to adjust heatmap color scheme and choose other SVs to display if uploaded data contains multiple cases.

### demo_data
- Check the official demo input [here](https://github.com/Nobel-Justin/Oviz-Bio-demo/blob/master/SV_Heatmap/demo_data/).

# Fusion Gene

## Fusion Trans Junction
the `Fusion: Trans Junction` visualization displays the whole structure of chimeric transcript structure.
### demo_data
- Check the official demo input [here](https://github.com/Nobel-Justin/Oviz-Bio-demo/blob/master/Fusion_Trans_Junction/demo_data/).

# Onco Virus

## Virus: Integ HotSpot
the `Virus: Integ HotSpot` visualization displays the oncovirus integrated hotspot of group samples in a genome browser way.
### demo_data
- Check the official demo input [here](https://github.com/Nobel-Justin/Oviz-Bio-demo/blob/master/Virus_IntegHotSpot/demo_data/).
- `HBV_PMID22634754.csv` and `HPV_PMID25581428.csv` record integration cases of oncovirus HBV and HPV, respectively.
